packages/xo-web/src/xo-app/new/network/index.js

```diff
@@ -32,23 +39,59 @@ const EMPTY = {
   isPrivate: false,
   mtu: '',
   name: '',
+  networks: [],
   pif: undefined,
   pifs: [],
   vlan: '',
 }
 
+const LineItem = ({ children }) => (
+  <div className={styles.lineItem}>{children}</div>
+)
+
+const Item = ({ label, children, className }) => (
+  <span className={styles.item}>
+    {label && (
+      <span>
+        {label}
+        &nbsp;
+      </span>
+    )}
+    <span className={classNames(styles.input, className)}>{children}</span>
+  </span>
+)
+
 const NewNetwork = decorate([
   addSubscriptions({
     plugins: subscribePlugins,
   }),
   connectStore(() => ({
     isPoolAdmin: getIsPoolAdmin,
+    nPools: createGetObjectsOfType('pool').count(),
     pool: createGetObject((_, props) => props.location.query.pool),
   })),
   injectIntl,
   provideState({
     initialState: () => ({ ...EMPTY, bondModes: undefined }),
     effects: {
+      addPool() {
+        const { state } = this
+        state.networks = [
+          ...state.networks,
+          { pool: undefined, pif: undefined },
+        ]
+      },
+      onChangeNetwork(_, key, pool, pif) {
```

Comment:
Why `onChangeNetwork`?

Reply:
Its when an element of the field `networks` changes.
I called it `networks` (and therefore `onNetworkChange`) because it represents the network created on the added pools to the root pool's network.